### ISSUE TITLE
Shell completion

### ISSUE THREAD

**Issue by yermulnik** [wimpysworld/deb-get#465]
## Requesting a new feature 🧑‍💻

Would be great to have Shell completion (for Bash at least) bundled into a `deb-get` package.
Is there an item in `deb-get`'s roadmap on adding such a feature?

### Expected behavior

Shell completion is installed along with `deb-get` package and users can utilise tab-completion for commands and parameters.

### Actual behavior

Shell completion is not install and hasn't been added to `deb-get` as of yet.

### Steps to reproduce

Install `deb-get` and run `deb-get<tab><tab>`

### System information

Run the following a paste the output below:

#### /etc/os-release

```bash
> cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.4 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.4 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal```

#### /etc/lsb-release

```bash
> cat /etc/lsb-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=20.04
DISTRIB_CODENAME=focal
DISTRIB_DESCRIPTION="Ubuntu 20.04.4 LTS"```

#### deb-get version

```bash
> deb-get version
0.3.5
```

**Comment by natanjunges** [comment]
I like this idea, anything that can save me a few key presses.

**Comment by natanjunges** [comment]
I came up with an implementation, but without the optimizations from https://github.com/natanjunges/deb-get/tree/extrepo, which is a WIP implementation for #313, it is too slow. I'll wait until #313 is finished and merged before submitting it.

**Comment by yermulnik** [comment]
@natanjunges Thank you for the time and effort 🙇🏻

**Comment by natanjunges** [comment]
Closed by #523.
